Orthtree/include/CGAL/Orthtree_traits_point_3.h

```diff
@@ -107,46 +108,55 @@ struct Orthtree_traits_point_3 : public Orthtree_traits_3_base<GeomTraits> {
-
-    typename Self::Array bbox_min;
-    typename Self::Array bbox_max;
-    Orthtrees::internal::Cartesian_ranges<Self> cartesian_range;
-
-    // init bbox with first values found
-    {
-      const typename Self::Point_d& point = get(m_point_map, *(m_point_set.begin()));
-      std::size_t i = 0;
-      for (const typename Self::FT& x: cartesian_range(point)) {
-        bbox_min[i] = x;
-        bbox_max[i] = x;
-        ++i;
+  auto root_node_bbox_object() const {
+    return [&]() -> std::pair<typename Self::Array, typename Self::Array> {
+
+      typename Self::Array bbox_min;
+      typename Self::Array bbox_max;
+      Orthtrees::internal::Cartesian_ranges<Self> cartesian_range;
+
+      // init bbox with first values found
+      {
+        const typename Self::Point_d& point = get(m_point_map, *(m_point_set.begin()));
+        std::size_t i = 0;
+        for (const typename Self::FT& x: cartesian_range(point)) {
+          bbox_min[i] = x;
+          bbox_max[i] = x;
+          ++i;
+        }
       }
-    }
-    // Expand bbox to contain all points
-    for (const auto& p: m_point_set) {
-      const typename Self::Point_d& point = get(m_point_map, p);
-      std::size_t i = 0;
-      for (const typename Self::FT& x: cartesian_range(point)) {
-        bbox_min[i] = (std::min)(x, bbox_min[i]);
-        bbox_max[i] = (std::max)(x, bbox_max[i]);
-        ++i;
+      // Expand bbox to contain all points
+      for (const auto& p: m_point_set) {
+        const typename Self::Point_d& point = get(m_point_map, p);
+        std::size_t i = 0;
+        for (const typename Self::FT& x: cartesian_range(point)) {
+          bbox_min[i] = (std::min)(x, bbox_min[i]);
+          bbox_max[i] = (std::max)(x, bbox_max[i]);
+          ++i;
+        }
       }
-    }
 
-    return {bbox_min, bbox_max};
+      return {bbox_min, bbox_max};
+    };
   }
 
-  typename Self::Node_data root_node_contents() const { return {m_point_set.begin(), m_point_set.end()}; }
+  auto root_node_contents_object() const {
+    return [&]() -> typename Self::Node_data {
+      return {m_point_set.begin(), m_point_set.end()};
+    };
+  }
 
-  template <typename Node_index, typename Tree>
-  void distribute_node_contents(Node_index n, Tree& tree, const typename Self::Point_d& center) {
-    CGAL_precondition(!tree.is_leaf(n));
-    reassign_points(tree, m_point_map, n, center, tree.data(n));
+  auto distribute_node_contents_object() const {
+    return [&](typename Tree::Node_index n, Tree& tree, const typename Self::Point_d& center) {
+      CGAL_precondition(!tree.is_leaf(n));
+      reassign_points(tree, m_point_map, n, center, tree.data(n));
+    };
   }
 
-  typename Self::Point_d get_element(const Node_data_element& index) const {
-    return get(m_point_map, index);
+  auto get_element_object() const {
+    return [&](const Node_data_element& index) -> typename Self::Point_d {
+      return get(m_point_map, index);
+    };
   }
```

Comment:
@sloriot I'm testing out switching over the new Traits functionality to the functor object approach used by `construct_bbox_d_object`. For now I've used lambda captures to avoid defining functors which need constructors, since they depend on State in the Traits class. 

This works fine as a temporary solution, but I will need to add more boilerplate in order to document the functor object itself. That should be pretty straightforward to add, but I would prefer not to go through with this change, because it makes defining a custom Traits class much more difficult for the user. Is there a document which explains why the functor+_object function idiom is necessary?